include/pso.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+double * pso_basic (double(*fit)(const double* const, size_t),
+                    size_t swarm_size,
+                    size_t dim, size_t max_iter,
+                    const double* const min_positions,
+                    const double* const max_positions);
+
+//returns swarm_size * dim size array
+double* pso_rand_init(size_t swarm_size,
+                  size_t dim,
+                  const double* const min_positions,
+                  const double* const max_positions);
+
+//returns swarm_size size array
+void pso_eval_fitness(double (*func)(const double* const, size_t),
+                  size_t swarm_size,
+                  size_t dim,
+                  const double* const positions,
+                  double* fitness);
+
+//returns swarm_size x dim size array
+// velocity at every posize_t of every particle
+double* pso_gen_init_velocity(const double* const positions,
+                          size_t swarm_size, size_t dim,
+                          const double* const min_positions,
+                          const double* const max_positions);
+
+// sorts the population in decreasing order of fitness /
+// i.e pop[0] posize_ts to best location yet.
+size_t pso_best_fitness(double* fitness,size_t dim, size_t swarm_size);
+
+
+void pso_generate_vel_limit(const double* min_positions,

```

Comment:
Short docstring description of the function would be nice in the header file. In the source file then either no doc string if everything is very clear or one going a little bit more in depth about implementation compared to the what-is-this-thing-doing docstring in the header file.

Reply:
sounds, good. 
will add these in the next PR. Plus the typos from the earlier two comments.